http3/roundtrip.go

```diff
@@ -66,6 +66,8 @@ type RoundTripOpt struct {
 	// no cached connection is available, RoundTrip
 	// will return ErrNoCachedConn.
 	OnlyCachedConn bool
+	// SkipSchemeCheck controls whether we check if the scheme is https.
+	SkipSchemeCheck bool
```

Comment:
How does Go's HTTP package deal with this? Does it check the scheme?

(In any case, the docs could be a bit more exhaustive – e.g. mention masque as the motivating use case.)

Reply:
It does, see https://github.com/golang/go/blob/520f3b72db7befab2028d9a47376267cf2d274a9/src/net/http/transport.go#L516 and
https://github.com/golang/go/blob/520f3b72db7befab2028d9a47376267cf2d274a9/src/net/http/transport.go#L546-L549. Only "http" and "https" are allowed. Good point on expanding the comment, will do.